addons/irlight/CfgWeapons.hpp

```diff
@@ -0,0 +1,393 @@
+    class Attenuation { \
+        constant = 1; \
+        linear = 0; \
+        quadratic = 0.02; \
+        start = 1; \
+        hardLimitStart = hardLimitStart; \
+        hardLimitEnd = hardLimitEnd; \
+    }
+#define SPIR_FLASHLIGHT_LR(hardLimitStart,hardLimitEnd) \
+    color[] = {1, 1, 1}; \
+    ambient[] = {1, 1, 1}; \
+    size = 1; \
+    position = "flash dir"; \
+    direction = "flash"; \
+    useFlare = 1; \
+    flareSize = 1.4; \
+    flareMaxDistance = 100; \
+    irLight = 1; \
+    volumeShape = "a3\data_f\VolumeLightFlashlight.p3d"; \
+    class Attenuation { \
+        constant = 1; \
+        linear = 0; \
+        quadratic = 0.002; \
+        start = 1; \
+        hardLimitStart = hardLimitStart; \
+        hardLimitEnd = hardLimitEnd; \
+    }
+#define POINTER \
+    class Pointer { \
+        irLaserPos = "laser pos"; \
+        irLaserEnd = "laser dir"; \
+        irDistance = 5; \
+    }
+
+class CfgWeapons {
+    class acc_pointer_IR;
+    class acc_flashlight;
+    class InventoryFlashlightItem_Base_F;
+
+    // DBAL-A3 (red pointer)
+    class ACE_DBAL_A3_Red: acc_pointer_IR {
+        author = ECSTRING(common,ACETeam);
+        displayName = CSTRING(DBAL_A3_Red);
+        descriptionUse = CSTRING(DBAL_A3_DescriptionUse);
+        descriptionShort = CSTRING(DBAL_A3_DescriptionShort);
+        MRT_SwitchItemNextClass = "ACE_DBAL_A3_Red_IP";
+        MRT_SwitchItemPrevClass = "ACE_DBAL_A3_Red_VP";
+        MRT_SwitchItemHintText = CSTRING(Mode_IRDual);
+        GVAR(baseClass) = "ACE_DBAL_A3_Red";
+
+        class ItemInfo: InventoryFlashLightItem_Base_F {
+            DBAL_A3_FLASHLIGHT;
+            POINTER;
+        };
+    };
+
+    class ACE_DBAL_A3_Red_IP: ACE_DBAL_A3_Red {
+        scope = 1;
+        MRT_SwitchItemNextClass = "ACE_DBAL_A3_Red_II";
+        MRT_SwitchItemPrevClass = "ACE_DBAL_A3_Red";
+        MRT_SwitchItemHintText = CSTRING(Mode_IRPointer);
+
+        class ItemInfo: InventoryFlashLightItem_Base_F {
+            class Flashlight {};
+            POINTER;
+        };
+    };
+
+    class ACE_DBAL_A3_Red_II: ACE_DBAL_A3_Red {
+        scope = 1;
+        MRT_SwitchItemNextClass = "ACE_DBAL_A3_Red_VP";
+        MRT_SwitchItemPrevClass = "ACE_DBAL_A3_Red_IP";
+        MRT_SwitchItemHintText = CSTRING(Mode_IRIlluminator);
+
+        class ItemInfo: InventoryFlashLightItem_Base_F {
+            DBAL_A3_FLASHLIGHT;
+        };
+    };
+
+    class ACE_DBAL_A3_Red_VP: ACE_DBAL_A3_Red {
+        scope = 1;
+        ACE_laserpointer = 1;
+        MRT_SwitchItemNextClass = "ACE_DBAL_A3_Red";
+        MRT_SwitchItemPrevClass = "ACE_DBAL_A3_Red_II";
+        MRT_SwitchItemHintText = CSTRING(Mode_VisiblePointer);
+
+        class ItemInfo: InventoryFlashLightItem_Base_F {
+            VISIBLE_POINTER_FLASHLIGHT;
+            POINTER;
+        };
+    };
+
+    // DBAL-A3 (red pointer, long range)
+    class ACE_DBAL_A3_Red_LR: ACE_DBAL_A3_Red {
```

Comment:
I'm stupid. How are these supposed to be accessed?

EDIT: I see the documentation now.